spanlogger/spanlogger_test.go

```diff
@@ -183,3 +200,103 @@ type noDebugNoopLogger struct{}
+			logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.Caller(6))
+			logger = dskit_log.NewRateLimitedLogger(logger, 1000, 1000, nil)
+			logger = level.NewFilter(logger, level.AllowAll())
+			return logger
+		},
+	}
+
+	resolver := fakeResolver{}
+
+	setupTest := func(t *testing.T, loggerFactory func(io.Writer) log.Logger) (*bytes.Buffer, *SpanLogger, *jaeger.Span) {
+		reporter := jaeger.NewInMemoryReporter()
+		tracer, closer := jaeger.NewTracer(
+			"test",
+			jaeger.NewConstSampler(true),
+			reporter,
+		)
+		t.Cleanup(func() { _ = closer.Close() })
+
+		span, ctx := opentracing.StartSpanFromContextWithTracer(context.Background(), tracer, "test")
+
+		buf := bytes.NewBuffer(nil)
+		logger := loggerFactory(buf)
+		spanLogger := FromContext(ctx, logger, resolver)
+
+		return buf, spanLogger, span.(*jaeger.Span)
+	}
+
+	requireSpanHasTwoLogLinesWithoutCaller := func(t *testing.T, span *jaeger.Span) {
+		logs := span.Logs()
+		require.Len(t, logs, 2)
+
+		require.Equal(t, []otlog.Field{otlog.String("msg", "this is a test")}, logs[0].Fields)
+		require.Equal(t, []otlog.Field{otlog.String("msg", "this is another test")}, logs[1].Fields)
+	}
+
+	for name, loggerFactory := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Run("logging with Log()", func(t *testing.T) {
+				logs, spanLogger, span := setupTest(t, loggerFactory)
+
+				_, thisFile, lineNumberTwoLinesBeforeFirstLogCall, ok := runtime.Caller(0)
+				require.True(t, ok)
+				_ = spanLogger.Log("msg", "this is a test")
+
+				logged := logs.String()
+				require.Contains(t, logged, "caller="+toCallerInfo(thisFile, lineNumberTwoLinesBeforeFirstLogCall+2))
+
+				logs.Reset()
+				_, _, lineNumberTwoLinesBeforeSecondLogCall, ok := runtime.Caller(0)
+				require.True(t, ok)
+				_ = spanLogger.Log("msg", "this is another test")
+
+				logged = logs.String()
+				require.Contains(t, logged, "caller="+toCallerInfo(thisFile, lineNumberTwoLinesBeforeSecondLogCall+2))
+
+				requireSpanHasTwoLogLinesWithoutCaller(t, span)
+			})
+
+			t.Run("logging with DebugLog()", func(t *testing.T) {
+				logs, spanLogger, span := setupTest(t, loggerFactory)
+				_, thisFile, lineNumberTwoLinesBeforeLogCall, ok := runtime.Caller(0)
+				require.True(t, ok)
+				spanLogger.DebugLog("msg", "this is a test")
+
+				logged := logs.String()
+				require.Contains(t, logged, "caller="+toCallerInfo(thisFile, lineNumberTwoLinesBeforeLogCall+2))
+
+				logs.Reset()
+				_, _, lineNumberTwoLinesBeforeSecondLogCall, ok := runtime.Caller(0)
+				require.True(t, ok)
+				spanLogger.DebugLog("msg", "this is another test")
+
+				logged = logs.String()
+				require.Contains(t, logged, "caller="+toCallerInfo(thisFile, lineNumberTwoLinesBeforeSecondLogCall+2))
+
+				requireSpanHasTwoLogLinesWithoutCaller(t, span)
+			})
+		})
+	}
+}
+
+func toCallerInfo(path string, lineNumber int) string {
+	fileName := path[strings.LastIndex(path, "/")+1:]
```

Comment:
Can't you use `filepath.Base` here? 

```suggestion
	fileName := filepath.Base(path)
```

Reply:
Yep, fixed in dc90524